src/openrct2/interface/Screenshot.cpp

```diff
@@ -320,6 +321,34 @@ static int32_t GetTallestVisibleTileTop(int32_t mapSize, int32_t rotation)
     return minViewY - 256;
 }
 
+static rct_drawpixelinfo CreateDPI(const rct_viewport& viewport)
+{
+    rct_drawpixelinfo dpi;
+    dpi.width = viewport.width;
+    dpi.height = viewport.height;
+    dpi.bits = new uint8_t[dpi.width * dpi.height]; //(uint8_t*)malloc((size_t)dpi.width * dpi.height);
+    if (dpi.bits == nullptr)
+    {
+        throw std::runtime_error("Giant screenshot failed, unable to allocate memory for image.");
+    }
```

Comment:
It makes sense to check the pointer if you're using ```malloc```, but if you're using ```new```, then you have to do a ```try...catch(std::bad_alloc& e)```, unless you use the `nothrow` version of `new`.